addons/website/tests/test_configurator_translation.py

```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from unittest.mock import patch
+
+import odoo.tests
+from odoo.addons.iap.tools.iap_tools import iap_jsonrpc_mocked
+from odoo.tools.misc import mute_logger
+
+@odoo.tests.common.tagged('post_install', '-at_install')
+class TestConfigurator(odoo.tests.HttpCase):
+
+    def _theme_upgrade_upstream(self):
+        # patch to prevent module install/upgrade during tests
```

Comment:
I was like "Cool, he thought about that", then saw it was a copy pasta.. You can't obviously copy paste a whole file to just add a few lines.. (in this case the if condition on language and the fr case)

You need to move this to the `test_configurator.py` file and use the same `setUp()`.
And what can't be shared inside the setup, you just move it to your test method.

Reply:
Fair enough. I was under the impression that you wanted the test to be inside website and not under the test_website_modules because it wasn't dependant of all the other website module, but rather just the configurator itself. I will move the file back.

Reply:
Yes that's true, I didn't notice it was in different module, but then you can just move the setup code from test_website_modules' test to the one you introduce, and makes test_website_modules class inherits from the one you introduce, so we don't end up with twice the same code.

Reply:
Any advice on how I could patch the inner funciton `iap_jsonrpc_mocked_configurator` from the original `test_configurator.py`?

Reply:
I am not too sure to understand what you mean.

The `test_configurator.py` will become a 8 lines file:

```python
# -*- coding: utf-8 -*-
# Part of Odoo. See LICENSE file for full copyright and licensing details.

from unittest.mock import patch

import odoo.tests
from odoo.addons.iap.tools.iap_tools import iap_jsonrpc_mocked
from odoo.addons.website.tests.test_configurator_translation import TestConfigurator


@odoo.tests.common.tagged('post_install', '-at_install')
class TestConfigurator(TestConfigurator):

    def test_01_configurator_flow(self):
        self.start_tour('/web#action=website.action_website_configuration', 'configurator_flow', login="admin")

```

Reply:
```diff
diff --git addons/test_website_modules/tests/test_configurator.py addons/test_website_modules/tests/test_configurator.py
index 972b2116d5f..e10cc13df6b 100644
--- addons/test_website_modules/tests/test_configurator.py
+++ addons/test_website_modules/tests/test_configurator.py
@@ -5,42 +5,11 @@ from unittest.mock import patch
 
 import odoo.tests
 from odoo.addons.iap.tools.iap_tools import iap_jsonrpc_mocked
+from odoo.addons.website.tests.test_configurator_translation import TestConfigurator
 
 
 @odoo.tests.common.tagged('post_install', '-at_install')
-class TestConfigurator(odoo.tests.HttpCase):
-
-    def _theme_upgrade_upstream(self):
-        # patch to prevent module install/upgrade during tests
-        pass
-
-    def setUp(self):
-        super().setUp()
-
-        def iap_jsonrpc_mocked_configurator(*args, **kwargs):
-            endpoint = args[0]
-            if endpoint.endswith('/api/website/1/configurator/industries'):
-                return {"industries": [
-                    {"id": 1, "label": "abbey"},
-                    {"id": 2, "label": "aboriginal and torres strait islander organisation"},
-                    {"id": 3, "label": "aboriginal art gallery"},
-                    {"id": 4, "label": "abortion clinic"},
-                    {"id": 5, "label": "abrasives supplier"},
-                    {"id": 6, "label": "abundant life church"}]}
-            elif '/api/website/2/configurator/recommended_themes' in endpoint:
-                return []
-            elif '/api/website/2/configurator/custom_resources/' in endpoint:
-                return {'images': {}}
-
-            iap_jsonrpc_mocked()
-
-        iap_patch = patch('odoo.addons.iap.tools.iap_tools.iap_jsonrpc', iap_jsonrpc_mocked_configurator)
-        iap_patch.start()
-        self.addCleanup(iap_patch.stop)
-
-        patcher = patch('odoo.addons.website.models.ir_module_module.IrModuleModule._theme_upgrade_upstream', wraps=self._theme_upgrade_upstream)
-        patcher.start()
-        self.addCleanup(patcher.stop)
+class TestConfigurator(TestConfigurator):
 
     def test_01_configurator_flow(self):
         self.start_tour('/web#action=website.action_website_configuration', 'configurator_flow', login="admin")
diff --git addons/website/tests/test_configurator_translation.py addons/website/tests/test_configurator_translation.py
index 16cad7d1152..a7cf1bd2e21 100644
--- addons/website/tests/test_configurator_translation.py
+++ addons/website/tests/test_configurator_translation.py
@@ -46,10 +46,6 @@ class TestConfigurator(odoo.tests.HttpCase):
 
             iap_jsonrpc_mocked()
 
-        with mute_logger('odoo.addons.base.models.ir_translation'):
-            self.env["base.language.install"].create({'lang': 'fr_FR', 'overwrite': True}).lang_install()
-        self.lang_id = self.env.ref('base.lang_fr')
-        self.fr_website_id = self.env['website'].create({'name': "Website Fr", 'default_lang_id': self.lang_id.id})
         iap_patch = patch('odoo.addons.iap.tools.iap_tools.iap_jsonrpc', iap_jsonrpc_mocked_configurator)
         iap_patch.start()
         self.addCleanup(iap_patch.stop)
@@ -59,7 +55,14 @@ class TestConfigurator(odoo.tests.HttpCase):
         self.addCleanup(patcher.stop)
 
     def test_01_configurator_translation(self):
+        with mute_logger('odoo.addons.base.models.ir_translation'):
+            self.env["base.language.install"].create({'lang': 'fr_FR', 'overwrite': True}).lang_install()
+        website_fr = self.env['website'].create({
+            'name': "Website Fr",
+            'default_lang_id': self.env.ref('base.lang_fr').id
+        })
+
         # disable configurator todo to ensure this test goes through
         active_todo = self.env['ir.actions.todo'].search([('state', '=', 'open')], limit=1)
         active_todo.update({'state': 'done'})
-        self.start_tour('/website/force/%s?path=%%2Fwebsite%%2Fconfigurator%%3Fdebug%%3Dassets' % self.fr_website_id.id, 'configurator_translation', login='admin')
+        self.start_tour('/website/force/%s?path=%%2Fwebsite%%2Fconfigurator%%3Fdebug%%3Dassets' % website_fr.id, 'configurator_translation', login='admin')
```

Reply:
I was thinking the other way around (inheriting TestConfigurator in TestConfiguratorTranslation), I miss read, this makes much more sense! Thanks!

Reply:
:) also maybe chose another class name, not the same one, name it like its file TestConfiguratorTranslation